objects/src/transaction/chain_mmr.rs

```diff
@@ -72,18 +73,18 @@ impl ChainMmr {
     }
 
     /// Returns total number of blocks contain in the chain described by this MMR.
-    pub fn chain_length(&self) -> usize {
-        self.mmr.forest()
+    pub fn chain_length(&self) -> BlockNumber {
+        BlockNumber::from_usize(self.mmr.forest())
```

Comment:
```suggestion
    pub fn chain_length(&self) -> BlockNumber {
        BlockNumber::from(
            u32::try_from(self.mmr.forest())
                .expect("chain mmr should never contain more than u32::MAX blocks"),
        )
    }
```

and then remove `BlockNumber::from_usize`. It just casts down to `u32` which could panic, so I don't think we should have this constructor.